### ISSUE TITLE
Upgrade neighbourhood shapes for Burlington, Vermont

### ISSUE THREAD

**Issue by wboykinm** [whosonfirst-data/whosonfirst-data#434]
## tl;dr

Per [this issue](https://github.com/whosonfirst/whosonfirst-sources/issues/8#issuecomment-218274557), zetashapes is a hot mess. 

## Status:

- [ ] Propose sources
- [ ] Establish concordances between WOF and source
- [ ] Upgrade / add WOF shapes
- [ ] Add new or missing features to WOF
- [ ] Deprecate / supersede WOF features
- [ ] Draft review
- [ ] Draft revisions based on review comments
- [ ] Final review
- [ ] Final revisions based on review comments
- [ ] Import into Who's On First

## Sourcing:

- Permissive license?
- Add new src to Who's On First

### Upgrade / add:

- macrohood
- neighbourhood
- microhood

### Add new or missing features to WOF

- macrohood
- neighbourhood
- microhood

### Deprecate / supersede:

- macrohood
- neighbourhood
- microhood

**Comment by wboykinm** [comment]
@nvkelso I have two blocks at the moment:

1. GeoJSON format is inscrutable. The repo source style doesn't match anything I usually compile to, so [the diffs end up looking ridiculous](https://github.com/whosonfirst-data/whosonfirst-data/pull/435/files), and it's not possible to interpret them.
2. I can't seem to find the background on the indexing scheme, so I can't add new features. Is there an accessible doc somewhere that tells me where I should put things in the file tree?

**Comment by stepps00** [comment]
Hey @wboykinm, thanks for the interest in Burlington, VT neighbourhoods! I'm going to jump in here to comment (and link to [our tutorial on updating Who's on First neighbourhoods](https://github.com/whosonfirst/whosonfirst-docs/wiki/Issue-%23125---A-Guide-to-Updating-Neighbourhood-Records-in-Who's-on-First-%28part-1%29)).

I went ahead and created a geojson file from [your diff](https://github.com/whosonfirst-data/whosonfirst-data/pull/435/files) - it looks like this is a geometry of the locality of Burlington, VT. 

We don't have existing neighbourhood, macrohood, or microhood records for Burlington, VT in Who's on First, so I would suggest compressing your geojson files into a .zip file and attaching them to a comment in this Issue. We can take your geojson files and import them independently from a PR.

@nvkelso may have another suggestion, but this would allow us to import new neighbourhood shapes that you suggest for Burlington.

**Comment by nvkelso** [comment]
Sounds right to me!

BTW, we do have a few neighbourhoods that are just outside the “city
limits”:
https://whosonfirst.mapzen.com/spelunker/id/102087451/descendants/?exclude=nullisland&placetype=neighbourhood#10/44.4458/-73.2131

**Comment by wboykinm** [comment]
Thanks @stepps00 - This is following up on a [zetashapes-is-a-mess issue](https://github.com/whosonfirst/whosonfirst-sources/issues/8) from awhile back. My original plan was to both clean the incorrect geometries and add the canonical ones (as well as clean up a few of the admin boundaries, as seen in the only commit to this PR so far). 

But man, it would be magnificent if you're willing to add the Burlington neighborhoods wholesale. Here's the cleaned geojson: https://gist.github.com/wboykinm/dfe44481d8ff759c4f1afea223a7c070 - note there aren't really any macrohoods - this is an itty-bitty city.

Thanks for the update guide! I'll dig through that for the surrounding neighborhoods that @nvkelso mentioned.

**Comment by stepps00** [comment]
No problem.. I took a look at the Burlington neighbourhoods and I have a few comments:

* It looks like there are both neighbourhoods and microhoods (see: [our placetypes repo](https://github.com/whosonfirst/whosonfirst-placetypes) in [your shapefile](https://gist.github.com/wboykinm/dfe44481d8ff759c4f1afea223a7c070). 

* _Neighbourhoods_ include: **Waterfront**, **North Hill section**, **South Hill section**, **Old North End**, **Riverside**, **Downtown**, **U. of Vermont**, **Intervale**, **New North End**, and **South End**.

* _Microhoods_ include: **Appletree Point**, **Lakewood** and **Van Patten** (all parented by New North End) and **Lakeside**, **The Five Sisters**, **The Addition**, and **Five Avenues** (all parented by South End).

* If you decide the above is correct, we'd have to follow the steps listed in the tutorial (create a source, compare and reconcile with current records, prepare attributes, and compile final geojson files for import). 

We welcome any contributions to the Who's on First records and we'll keep this issue open until we've finalized/imported new neighbourhood records for Burlington!

**Comment by wboykinm** [comment]
@stepps00 That looks correct - I see that I had misclassified the Riverside neighborhood in the `type` attribute - I've just updated it.

**Comment by stepps00** [comment]
The shapes look good! Since we don't have to reconcile the neighbourhood and microhood shapes in Burlington, VT with existing records (we don't have any in Who's on First), I can prepare for import fairly quickly.  A few to-do items:

* Create a source file in our [sources repo](https://github.com/whosonfirst/whosonfirst-sources/tree/master/sources) with relevant license and URLs.
* Check out the [neighboring records to Burlington](https://whosonfirst.mapzen.com/spelunker/id/102087451/descendants/?exclude=nullisland&placetype=neighbourhood#10/44.4458/-73.2131) to make sure they don't _actually_ fall within Burlingaton.
* Add a `wof:id` field and `parented_by` field to attributes table for neighbourhood and microhood records.

**Comment by stepps00** [comment]
@wboykinm - I looked over a few of the [surrounding neighbourhoods](https://whosonfirst.mapzen.com/spelunker/id/85688763/descendants/?exclude=nullisland&placetype=neighbourhood#11/44.4818/-73.2136) outside of Burlington, VT and a couple of them (like [Starr Farm Beach](https://whosonfirst.mapzen.com/spelunker/id/85850949/#11/44.5103/-73.2913) and [Kirby Corner](https://whosonfirst.mapzen.com/spelunker/id/85828531/#11/44.4451/-73.1289)) look like they fall within the boundaries proposed in your shapefile. 

They seem large and overlap the boundary of Burlington, so we'll have to decide on an action plan for them. Do you have any knowledge of these neighbourhoods?

**Comment by wboykinm** [comment]
@stepps00 Yes - those neighborhoods are actually completely wrong - per a [previous discussion w/ @nvkelso](https://github.com/whosonfirst/whosonfirst-sources/issues/8#issuecomment-218274557), we found that zetashapes was compiling geometries with no rhyme or reason. 

- Starr Farm Beach is not a recognized Burlington neighborhood (it's a park)
- Kirby corner is not in Burlington (It's in South Burlington, which is a different city)
- Similarly, [Twin Orchards](https://whosonfirst.mapzen.com/spelunker/id/85852901/#11/44.4184/-73.2388) is a neighborhood in South Burlington that has been erroneously mapped onto Burlington.

In the latter two cases, the geometries are so wildly inaccurate that they should be thrown out and begun fresh. It's not clear to me if those were maliciously hand-drawn or if a segmentation algorithm went crazy, but in either case zetashapes does not appear to be accepting corrections, which is why I started with WOF.

**Comment by nvkelso** [comment]
Are the quattroshapes alt geometries better for these? Better maybe, but still looks pretty wrong to me sitting 3k miles away!

<img width="728" alt="screen shot 2016-08-09 at 19 02 55" src="https://cloud.githubusercontent.com/assets/853051/17539582/19cee366-5e64-11e6-8ea6-7b8a44b38bce.png">


For features like **Starr Farm Beach** we have an option to add them as a `neighbourhood` (usually the cutoff is a couple hundred acres in size), but mark them as `mz:is_landuse_aoi` = `1`. That way there aren't holes in the neighbourhood coverage when the part isn't obviously part of the neighbouring neighbourhooods.

**Comment by wboykinm** [comment]
@nvkelso Quattroshapes is still way off. This is the approximate Twin Orchards location:

![orchards](https://www.dropbox.com/s/r6ndp7al07ve3gh/Screenshot%202016-08-10%2000.12.43.png?dl=1)

I say "approximate" because the city of South Burlington does not define neighborhood boundaries, and there has been no project to crowdsource them. [Unnecessary local history aside: [two orchards much smaller than the ungainly blob above] (http://wboykinm.github.io/btv-1937/#17/44.43816/-73.21010) were developed into housing in the 50s. They even named the streets after apple varieties.]

The inaccuracy of quattroshapes here is at least attributable to the geotagged-photo methodology working poorly in more sparsely-populated areas. But I still have no idea why zetashapes is so inaccurate. Looking at both sources in the Burlington metro area I can't say that __any__ of the features are spatially accurate, and at least half __of the names__ do not exist in a sense recognizable by their residents. I can't stress this enough: __These polygons are not reasonably-interpreted ideas of neighborhoods. These are machine-generated bugs, and they should be removed before someone actually tries to navigate to "Colchester" with a WOF-powered app and is directed to [this seasonally-inundated island](https://whosonfirst.mapzen.com/spelunker/id/85811655/#14/44.5233/-73.2137) instead.__

![chitt](https://www.dropbox.com/s/kh2d1btry3dy5a6/Screenshot%202016-08-09%2023.47.40.png?dl=1)

Neither zetashapes nor quattroshapes are even a little bit reliable for location-based queries in the entire region. I'd like to fix that here, starting with known boundaries available for the city of Burlington. 

But it also calls into question the intent of these sources beyond major metro areas. What does WOF gain by including data that is __known__ to be incorrect, at least in this case? How many other rural reaches and small cities have this problem as well?

**Comment by nvkelso** [comment]
It's a mixed bag. 

In the areas of California I'm more familiar with I think most the ~15,000 `mz:tier_local`=`1` _polygons_ are not good, but many of the _place names_ should be kept as a point label only (they are certainly present in other data products, or if they have been removed it's because someone weeded them out since they come from federal GNIS sources). Most of the features around you in Vermont seem to be in this bucket, sorry :\

Of the ~10,000 some `mz:tier_local`=`2` polygons those seem to be for more common places (with some trash, sure) but with a grossly huge polygon and we should probably go back to the quattroshapes polygon as default geometry (instead of zetashapes). 

Tier 3 and 4 mostly seem good, but sometimes with oversized polygons. The nice thing about zetashapes is they follow census geographies so inside city limits make more sense, generally speaking.

**Comment by wboykinm** [comment]
@stepps00 I haven't forgotten that [these items are still on my list](https://github.com/whosonfirst-data/whosonfirst-data/issues/434#issuecomment-238603406)

**Comment by nvkelso** [comment]
The wof:id. It's  is the common currency of all of Who's On First :)

**Comment by stepps00** [comment]
@wboykinm - No problem, thanks for the contribution! Let me know if need anything to get the ball rolling.

And, yes - the `wof:id` is our stable identifier for Who's on First.

**Comment by wboykinm** [comment]
@stepps00 Many apologies, but I've lost the doc reference: how do I assign a wof:id? Is it just like an arbitrary uuid or something?

**Comment by thisisaaronland** [comment]
http://brooklynintegers.com/create :D 

Or this: http://brooklynintegers.com/api/

**Comment by stepps00** [comment]
@wboykinm - No worries, [here](https://github.com/whosonfirst/whosonfirst-cookbook/blob/master/definition/brooklyn_integers.md) is a brief write-up that defines **Brooklyn Integers**, the service you can use to create a new `wof:id`. 

When you create a new id, just copy it to your feature or source; it will be unique to that feature or source.

**Comment by wboykinm** [comment]
AWESOOOOOOME

**Comment by wboykinm** [comment]
@stepps00 I've updated [the source](https://gist.github.com/wboykinm/dfe44481d8ff759c4f1afea223a7c070) with `wof:id` and `parented_by` - according to the stock list above the next step is to flag the overlapping zetashapes features for deletion. What's the recommended way to do that?

**Comment by stepps00** [comment]
@wboykinm  - we have a write-up about what to do with the existing "bad" shapes in Who's On First once you've prepared a new dataset for import. See **4: Reconciling our data sources** in our [neighbourhood tutorial](https://github.com/whosonfirst/whosonfirst-cookbook/blob/master/issue_workflows/sf_neighbourhood_updates_pt_1.md).

If the feature was never a neighbourhood to begin with:
**1.** Update the `edtf:deprecated` field with today's date (2016-09-19)
**2.** Add a `mz:is_current` field with a value of `0`

If the feature was at one point valid, but should now be removed from our database:
**1.** Add a `edtf:cassation` field with today's date (2016-09-19)
**2.** Add a `mz:is_current` field with a value of `0`
**3.** Add the `wof:id` of the new feature that takes place of this "bad" feature in the "bad" feature's `superseded_by` field

Let me know if this makes sense. Your attribute fields may be labelled slightly differently, but the above are the general workflows for deprecating/adding cessation dates to features.
